azure/durable_functions/models/DurableEntityContext.py

```diff
@@ -0,0 +1,219 @@
+
+        Returns
+        -------
+        bool
+            True if the Entity was newly constructed. False otherwise.
+        """
+        # TODO: not updating this atm
+        return self._is_newly_constructed
+
+    @classmethod
+    def from_json(cls, json_str: str) -> Tuple['DurableEntityContext', List[Dict[str, Any]]]:
+        """Instantiate a DurableEntityContext from a JSON-formatted string.
+
+        Parameters
+        ----------
+        json_string: str
+            A JSON-formatted string, returned by the durable-extension,
+            which represents the entity context
+
+        Returns
+        -------
+        DurableEntityContext
+            The DurableEntityContext originated from the input string
+        """
+        json_dict = json.loads(json_str)
+        json_dict["name"] = json_dict["self"]["name"]
+        json_dict["key"] = json_dict["self"]["key"]
+        json_dict.pop("self")
+
+        serialized_state = json_dict["state"]
+        if serialized_state is not None:
+            json_dict["state"] = json.loads(serialized_state,
+                                            object_hook=_deserialize_custom_object)
+        batch = json_dict.pop("batch")
+        return cls(**json_dict), batch
+
+    def set_state(self, state: Any) -> None:
+        """Set the state of the entity.
+
+        Parameter
+        ---------
+        state: Any
+            The new state of the entity
+        """
+        # TODO: enable serialization of custom types
+        self._exists = True
+        self._state = json.dumps(state)
+
+    def get_state(self, initializer: Optional[Callable[[], Any]] = None) -> Any:
+        """Get the current state of this entity.
+
+        Parameters
+        ----------
+        initializer: Optional[Callable[[], Any]]
+            A 0-argument function to provide an initial state. Defaults to None.
+
+        Returns
+        -------
+        Any
+            The current state of the entity
+        """
+        state = self._state
+        # TODO: some weird errs here with None states
+        if state is not None:
+            return state
+        elif initializer:
+            # TODO: ensure this is a fucntion
+            state = initializer()
+        return state
+
+    def get_input(self) -> Any:
+        """Get the input for this operation.
+
+        Returns
+        -------
+        Any
+            The input for the current operation
+        """
+        input_ = None
+        req_input = self._input
+        req_input = json.loads(req_input)
+        input_ = None if req_input is None else self.from_json_util(req_input)
+        return input_
+
+    def set_result(self, result: Any) -> None:
+        """Set the result (return value) of the entity.
+
+        Paramaters
+        ----------
+        result: Any
+            The result / return value for the entity
+        """
+        self._exists = True
+        self._result = result
+
+    def destruct_on_exit(self) -> None:
```

Comment:
I'm not sure how we handle this in JS or C#, but there is the question of how to handle subsequent operations after this within a batch.

Reply:
Let's check-in with the team then 😉 
@cgillum , @sebastianburckhardt:

What are the semantics for a `destructed` entity (via destructOnExit) when there are still operations to perform on a batch?

Reply:
I'll be tracking this here: https://github.com/Azure/azure-functions-durable-python/issues/222